morf-testsupport/src/main/java/org/alfasoftware/morf/testing/UpgradeTestHelper.java

```diff
@@ -188,42 +197,63 @@ public void validateUpgradeStepProperties(Iterable<Class<? extends UpgradeStep>>
-        upgradeStepClass.getSimpleName(), java.util.UUID.randomUUID().toString()));
-    }
-
-    // verify we can parse the UUID
-    try {
-      assertNotNull(java.util.UUID.fromString(currentUuid));
-    } catch (Exception e) {
-      throw new RuntimeException(String.format("Could not parse UUID [%s] from [%s]", currentUuid, upgradeStepClass.getSimpleName()), e);
+      errors.add(String.format("Upgrade step [%s] must have a non blank, unique UUID set. How about [%s]?",
+        upgradeStepClass.getSimpleName(), java.util.UUID.randomUUID()));
+    } else {
+      // verify we can parse the UUID
+      try {
+        assertNotNull(java.util.UUID.fromString(currentUuid));
+      } catch (Exception e) {
+        String errorMessage = String.format("Could not parse UUID [%s] from [%s]", currentUuid, upgradeStepClass.getSimpleName());
+        LOG.error(errorMessage, e);
+        errors.add(errorMessage);
+      }
     }
 
     // Check the upgrade step has a description
     final String description = upgradeStep.getDescription();
-    assertTrue(String.format("[%s] should have a description", upgradeStepClass.getSimpleName()), StringUtils.isNotEmpty(description));
-    assertTrue(String.format("Description for [%s] must not be more than 200 characters", upgradeStepClass.getSimpleName()), description.length() <= 200);
-    assertFalse(String.format("Description for [%s] should not end with full stop", upgradeStepClass.getSimpleName()), description.endsWith("."));
 
-    assertTrue(String.format("[%s] should have a JIRA ID", upgradeStepClass.getSimpleName()), StringUtils.isNotEmpty(upgradeStep.getJiraId()));
+    if (StringUtils.isEmpty(description)) {
+      errors.add(String.format("Upgrade step [%s] must have a description", upgradeStepClass.getSimpleName()));
+    } else {
+      if (description.length() > 200) {
+        errors.add(String.format("Description for [%s] must not be more than 200 characters", upgradeStepClass.getSimpleName()));
+      }
+      if (description.endsWith(".")) {
+        errors.add(String.format("Description for [%s] must not end with full stop", upgradeStepClass.getSimpleName()));
+      }
+    }
 
-    for (String jiraId : StringUtils.split(upgradeStep.getJiraId(), ',')) {
-      assertTrue(String.format("[%s] should have a valid JIRA ID [%s]", upgradeStepClass.getSimpleName(), upgradeStep.getJiraId()), jiraIdIsValid(jiraId));
+    if (StringUtils.isEmpty(upgradeStep.getJiraId())) {
+      errors.add(String.format("Upgrade step [%s] must have a JIRA ID", upgradeStepClass.getSimpleName()));
+    } else {
+      for (String jiraId : StringUtils.split(upgradeStep.getJiraId(), ',')) {
+        if (!jiraIdIsValid(jiraId)) {
+          errors.add(String.format("Upgrade step [%s] must have a valid JIRA ID [%s]", upgradeStepClass.getSimpleName(), upgradeStep.getJiraId()));
+        }
+      }
     }
+
+    return errors;

```

Comment:
This returns a mutable ArrayList.
Maybe use ImmutableList instead.
Or wrap this in Collections.unmodifiableList(list)